### PR TITLE
feat(module: select): add EnumSelect for select componet with enum

### PR DESCRIPTION
**Pull request by wangj90**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  add EnumSelect for select componet with enum  |
| 🇨🇳 Chinese |  增加EnumSelect，用于带有枚举的Select组件  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/ant-design-blazor/ant-design-blazor?pullRequest=1759) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/ant-design-blazor/ant-design-blazor?pullRequest=1759) before we can accept your contribution.<br/><br/>:x: wangj90 [sign now](https://cla.dotnetfoundation.org/ant-design-blazor/ant-design-blazor?pullRequest=1759)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/ant-design-blazor/ant-design-blazor?pullRequest=1759) it.</sub>

**Comment by anddrzejb** [comment]
It would be beneficial to the community, if you could add a demo to the docs.

**Comment by wangj90** [comment]
> It would be beneficial to the community, if you could add a demo to the docs.

Now with added demo to the docs.

File: components/select/EnumSelect.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace AntDesign
+{
+    public class EnumSelect<TEnum> : Select<TEnum, EnumLabelValue<TEnum>>
+    {
+        public EnumSelect()
+        {
+            if (typeof(TEnum).IsEnum)
+            {
+                DataSource = EnumHelper<TEnum>.GetLabelValueList();
```
Comment (anranruye) [inline components/select/EnumSelect.cs:13]: We have implemented simple data source in #1541. So here you can use 
```C#
public class EnumSelect<TEnum> : Select<TEnum, TEnum>
...
            DataSource = Enum.GetValues(typeof(T)).Cast<T>();
...
```

**Review by ElderJames — APPROVED** [review]

**Pull request merged**